pkg/partmgr/partmgr.go

```diff
@@ -18,123 +18,201 @@ under the Apache 2.0 license is conditioned upon your compliance with
+	}
+	err = p.container.Sync.PutObject(&v3io.PutObjectInput{Path: p.path + config.SCHEMA_CONFIG, Body: data})
+	return err
+}
 
-	return p.headPartition
+func (p *PartitionManager) PartsForRange(mint, maxt int64) []*DBPartition {
+	var parts []*DBPartition
+	for _, part := range p.partitions {
+		if part.startTime >= mint && (maxt == 0 || part.startTime < maxt) {
+			parts = append(parts, part)
+		}
+	}
+	return parts
 }
 
 type DBPartition struct {
-	manager        *PartitionManager
-	path           string             // Full path (in the DB) to the partition
-	partID         int                // PartitionID
-	startTime      int64              // Start from time/date
-	days           int                // Number of days stored in the partition
-	hoursInChunk   int                // number of hours stored in each chunk
-	prefix         string             // Path prefix
-	retentionDays  int                // Keep samples for N days
-	defaultRollups aggregate.AggrType // Default Aggregation functions to apply on sample update
-	rollupTime     int64              // Time range per aggregation bucket
-	rollupBuckets  int                // Total number of buckets per partition
+	manager           *PartitionManager
+	path              string             // Full path (in the DB) to the partition
+	startTime         int64              // Start from time/date
+	partitionInterval int64              // Number of millis stored in the partition
+	chunkInterval     int64              // number of millis stored in each chunk
+	prefix            string             // Path prefix
+	retentionDays     int                // Keep samples for N hours
+	defaultRollups    aggregate.AggrType // Default Aggregation functions to apply on sample update
+	rollupTime        int64              // Time range per aggregation bucket
+	rollupBuckets     int                // Total number of buckets per partition
 }
 
 func (p *DBPartition) IsCyclic() bool {
 	return p.manager.cyclic
 }
 
-func (p *DBPartition) HoursInChunk() int {
-	return p.hoursInChunk
+// Time covered by a single chunk
+func (p *DBPartition) TimePerChunk() int64 {
+	return p.chunkInterval
 }
 
-func (p *DBPartition) NextPart(t int64) *DBPartition {
+func (p *DBPartition) NextPart(t int64) (*DBPartition, error) {
 	return p.manager.TimeToPart(t)
 }
 
-func (p *DBPartition) GetId() int {
-	return p.partID
+func (p *DBPartition) GetStartTime() int64 {
+	return p.startTime
 }
 
-// return path to metrics table and filter expression for selected partition
-func (p *DBPartition) GetTablePath() (string, string) {
-	return p.path, ""
+func (p *DBPartition) GetEndTime() int64 {
+	return p.startTime + p.partitionInterval
+}
+
+// return path to metrics table
+func (p *DBPartition) GetTablePath() string {
+	return p.path
+}
+
+// return list of Sharding Keys matching the name
+func (p *DBPartition) GetShardingKeys(name string) []string {
+	var res []string
```

Comment:
you can initialise the slice with `len(p.manager.cfg.TableSchemaInfo.ShardingBuckets)`